src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -145,11 +145,6 @@ public int[] getAcceptableTokens() {
         return getRequiredTokens();
     }
 
-    @Override
-    public void beginTree(DetailAST ast) {
-        defined = false;
```

Comment:
reminder, all initialization in this method, is reset of Check state before starting new file, to reproduce problem we need to run same Check instance on 2 special input files to let "defined = true" affect validation in second file even it has no java code that make "defined = true".

Reply:
Do not deal with pure Junit method , we should cover it by Inputs.
New or existing verify method that takes creates check ones and execute two files sequentially.

Reply:
@romani, I have tried to run 2 files sequentially but the only thing changed here is if I make https://github.com/checkstyle/checkstyle/blob/24a4b06182f9b9066a684715f324d2bbf7099d10/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java#L148-L151 
defined = true then my test case will fail but it is still not resolving our actual mutation which is to remove that line 
`defined = false`

Reply:
Done